tests/pytests/test_numbers.py

```diff
@@ -389,3 +389,69 @@ def testNegativeValues(env):
+    conn = getConnectionByEnv(env)
+
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'n', 'numeric').ok()
+    env.assertEqual(conn.execute_command('HSET', 'doc01', 'n', '1.0'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc02', 'n', '1'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc03', 'n', '1.0e0'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc04', 'n', '10e+2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc05', 'n', '1.5e+2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc06', 'n', '10e-2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc07', 'n', '1.5e-2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc08', 'n', 'INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc09', 'n', 'inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc10', 'n', 'iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc11', 'n', '+INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc12', 'n', '+inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc13', 'n', '+iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc14', 'n', '-INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc15', 'n', '-inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc16', 'n', '-iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc17', 'n', '-1'), 1)
+
+    # Test unsigned numbers
+    expected = [3, 'doc01', ['n', '1.0'], 'doc02', ['n', '1'],
+                'doc03', ['n', '1.0e0']]
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1 1]')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1e0 1]')
+    env.assertEqual(res, expected)
+
+    # Test signed numbers
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[+1e0 +1]')
+    env.assertEqual(res, expected)
+
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[-1e0 -1]')
+    env.assertEqual(res, [1, 'doc17', ['n', '-1']])
+
+    # Test +inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:[1e6 inf]')
+    expected = [6, 'doc08', ['n', 'INF'], 'doc09', ['n', 'inf'],
+                'doc10', ['n', 'iNf'], 'doc11', ['n', '+INF'],
+                'doc12', ['n', '+inf'], 'doc13', ['n', '+iNf']]
+    env.assertEqual(res1, expected)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[1e6 INF]')
+    env.assertEqual(res2, expected)
+
+    # Test -inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:[-inf 0]')
+    expected = [4, 'doc14', ['n', '-INF'], 'doc15', ['n', '-inf'],
+                'doc16', ['n', '-iNf'], 'doc17', ['n', '-1']]
+    env.assertEqual(res1, expected)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[-INF 0]')
+    env.assertEqual(res2, expected)
+
+    # Multiple parenthesis are allowed
+    expected = [1, 'doc05', ['n', '1.5e+2']]
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1e2 ((300]')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[((1 300]')
+    env.assertEqual(res, expected)
+
+    # invalid syntax - multiple signs are not allowed
```

Comment:
Test also multiple signs with `inf`, e.g., `--inf`

And also when using a parameter, e.g.,
`-$param` when param is `-inf` - should fail
`-$param` when param is `inf` - should become `-inf`?

Reply:
New tests added in `test_numbers.py` and `test_search_params.py`

I think that `-$param` when param is `-inf` - should be valid, because we are replacing the numeric value and after that applying the unary operator, it is the same case that `-$param` when param is `+3` it is valid and becomes `-3`

I modified the range limits validation of `+/-Inf` and the corresponding test in `test.py`, please check if it is correct.

Reply:
Parameters would behave differently than plain values, for example, `--inf` fails but `-$param` succeeds when param is `-inf`, but since parameter substitutions occur in a different phase, it does make sense to support such valid parameter substitution use cases.